editor/script/store.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * ```js
+ * // call once
+ * Store.init(function onFirstError() {
+ * 	// called once when storage fails;
+ * 	// use to signal lack of storage to user
+ * 	window.alert('Storage is not working!');
+ * });
+ * // call as needed
+ * Store.get('key', 'fallback value');
+ * Store.set('key', 'new value');
+ * ```
+ */
+var Store = {
+	/** storage driver (must implement `getItem`, `setItem`, `removeItem`) */
+	driver: localStorage,
+	/** whether storage has failed */
+	error: false,
+	/**
+	 * Initializes storage mechanism
+	 * @param {(err: Error) => void} [onFirstError] called once when storage fails;
+	 * use to signal lack of storage to user
+	 */
+	init: function (onFirstError) {
+		this.onFirstError = function(err) {
+			this.error = true;
+			console.warn('Storage error', err);
+			if (onFirstError) onFirstError(err);
+		};
+		try {
+			// test that storage driver is available and working
+			this.driver.setItem('_test_key_', 'test');
```

Comment:
```suggestion
			this.driver = localStorage;
			var testValue = 'test';
			this.driver.setItem('_test_key_', testValue);
			if (this.driver.getItem('_test_key_') !== testValue) {
				throw('Storage access fails silently. This might be caused by ' +
					  'a browser extension that blocks third-party cookies.'
				);
			}
```